influx-reports/influx.py

```diff
@@ -1,45 +1,87 @@
-#!/usr/bin/env python3
-""" Get all measurements for an influx job, and
-outputs a report of uptime for maas_ components.
-Inputs:
-    - env.INFLUX_IP (mandatory)
-    - env.INFLUX_PORT (mandatory)
-    - env.BUILD_TAG (mandatory)
-    - ymlreport <ymlreportfile> (optional)
-    - subunitreport <subunitreport> (optional)
-Outputs:
-    - stdout
-    - yaml file with uptime (when ymlreport is used)
-    - subunit binary file with passing/failure criteria
-    (when subunitreport is used)
-"""
+#!/usr/bin/env python
 
 import argparse
 from collections import defaultdict
-import datetime
+from copy import deepcopy
+from datetime import datetime, timedelta
+from glob import iglob
 import logging
 import os
-import math
 import sys
-import yaml
 
-import dateutil.parser
+import dateutil
 from influxdb import InfluxDBClient
+import pytz
 from subunit.v2 import StreamResultToBytes
+import yaml
+
+
+LEAPFROG_STAGES = [
```

Comment:
@hughsaunders this section associates marker files generated by the upgrade process with scripts that are run to perform the upgrade. Each marker file is used to get the time when that part of the upgrade completes. This allows the Telegraf data to be associated with particular parts of the upgrade.